src/realtime_pollen_calibration/utils.py

```diff
@@ -1 +1,552 @@
+        if pollen_type != "POAC":
+            if change_tthre_saisl[istation] > 0:
+                change_tthre_saisl[istation] = min(
+                    failsafe[pollen_type], change_tthre_saisl[istation]
+                )
+            elif change_tthre_saisl[istation] <= 0:
+                change_tthre_saisl[istation] = max(
+                    -failsafe[pollen_type], change_tthre_saisl[istation]
+                )
+        else:  # POAC
+            if change_tthre_saisl[istation] > 0:
+                change_tthre_saisl[istation] = min(7, change_tthre_saisl[istation])
+            elif change_tthre_saisl[istation] <= 0:
+                change_tthre_saisl[istation] = max(-7, change_tthre_saisl[istation])
+        if verbose:
+            print(
+                f"Change tthrs is now {change_tthrs[istation]} ",
+                f"and change tthre/saisl is now {change_tthre_saisl[istation]}",
+            )
+            print("-----------------------------------------")
+    return change_tthrs, change_tthre_saisl
+
+
+def to_grib(inp, outp, dict_fields):
+    """Output fields to a GRIB file.
+
+    Args:
+        inp: Location of the GRIB file which must contain at least the same
+    fields as the ones in the dictionary that are to be outputted.
+        outp: Location of the desired GRIB file
+        dict_fields: Dictionary containing the fields to be outputted as
+            { name : value }
+
+    """
+    # copy all the fields from input into output,
+    # besides the ones in the dictionary given as input
+    with open(inp, "rb") as fin, open(outp, "wb") as fout:
+        while 1:
+            gid = eccodes.codes_grib_new_from_file(fin)
+            if gid is None:
+                break
+            # clone record
+            clone_id = eccodes.codes_clone(gid)
+            # get short_name
+
+            short_name = eccodes.codes_get_string(clone_id, "shortName")
+            # read values
+            values = eccodes.codes_get_values(clone_id)
+            eccodes.codes_set(
+                clone_id, "dataTime", eccodes.codes_get(clone_id, "dataTime") + 100
+            )
+            if short_name in dict_fields:
+                eccodes.codes_set_values(clone_id, dict_fields[short_name].flatten())
+            else:
+                eccodes.codes_set_values(clone_id, values)
+
+            eccodes.codes_write(clone_id, fout)
+            eccodes.codes_release(clone_id)
+            eccodes.codes_release(gid)
+        fin.close()
+        fout.close()
+
+
+def get_pollen_type(ds):
+    """Get the pollen type from the variables in the xarray.DataSet.
+
+    Args:
+        ds: xarray.DataSet containing one pollen-related field.
+
+    Returns:
+        pollen_type: String from pollen_types.
+
+    """
+    pollen_types = ["ALNU", "BETU", "POAC", "CORY"]
+    for var in ds:
+        if var[:4] in pollen_types:
+            return var[:4]
+    return "Error: pollen type not recognized."
```

Comment:
As mentioned before, I don't think this will work when there are multiple species modelled at the same time, or will it? We will have to make sure that this runs robustly for 1-3 species modelled simultaneously. Either calling this python routine once for all three species or just calling it three times, once for each species seperately.